### ISSUE TITLE
`stack --no-nix-pure` should not imply `--nix`

### ISSUE THREAD

**Issue by fmthoma** [commercialhaskell/stack#2825]
I have a Haskell script I want to run with stack as a script interpreter. This script should run both on NixOS and on non-Nix systems. The script depends on the locale, which is unset by the pure environment, therefore I want to run the script in a non-pure environment if Nix is enabled, but adding `--no-nix-pure` also enables `--nix` if `stack` isn't configured to use nix, breaking the script on non-Nix systems.

### Steps to reproduce

`~/.stack/config.yaml`:
```
…
nix:
  enable: false
```

`echo.hs`:
```
#!/usr/bin/env stack
-- stack --no-nix-pure runhaskell

main :: IO ()
main = getLine >>= putStrLn
```

Running this on a non-Nix system produces:
```
> echo äöü | ./echo.hs
Executable named nix-shell not found on path: [ … ]
```

Removing the `--no-nix-pure` on a Nix system (`nix: enable: true`) breaks the script because the default locale in a pure environment does not support UTF-8 (see also #2358 for the locale problem):
```
> echo äöü | ./echo.hs
echo.hs: <stdin>: hGetLine: invalid argument (invalid byte sequence)
```

### Expected

```
> echo äöü | ./echo.hs
äöü
```
on both Nix and non-Nix systems.

I would expect `--no-nix-pure` to switch to non-pure nix build if `--nix` is enabled, not enabling `--nix` in the same turn.
The current behavior makes it impossible to use `stack` as a script interpreter in a way compatible with both Nix and non-Nix systems.

### Stack version

```
$ stack --version
Version 1.2.0, Git revision 241cd07d576d9c0c0e712e83d947e3dd64541c42 (4054 commits) x86_64 hpack-0.14.0
```

**Comment by fmthoma** [comment]
Possible fix:

```diff
diff --git a/src/Stack/Options/NixParser.hs b/src/Stack/Options/NixParser.hs
index f1b0fcf..5db1570 100644
--- a/src/Stack/Options/NixParser.hs
+++ b/src/Stack/Options/NixParser.hs
@@ -10,7 +10,7 @@ import           Stack.Options.Utils
 import           Stack.Types.Nix
 
 nixOptsParser :: Bool -> Parser NixOptsMonoid
-nixOptsParser hide0 = overrideActivation <$>
+nixOptsParser hide0 =
   (NixOptsMonoid
   <$> pure (Any False)
   <*> firstBoolFlags nixCmdName
@@ -50,7 +50,4 @@ nixOptsParser hide0 = overrideActivation <$>
   )
   where
     hide = hideMods hide0
-    overrideActivation m =
-      if m /= mempty then m { nixMonoidEnable = (First . Just . fromFirst True) (nixMonoidEnable m) }
-      else m
     textArgsOption = fmap (map T.pack) . argsOption
```

I get why this was introduced in the first place, but keep in mind that this makes it nearly impossible to write `stack`-interpreted scripts that work in Nix environments just the same way as in non-Nix environments (the same problem holds for packages: If my script depends on some packages, I may want to add `--nix-packages …` to the `-- stack runghc` line without breaking the build on non-Nix machines). I think explicitly having to specify `--nix` in order to enable a Nix build is a small price to pay. As a NixOS user, I have `nix: enable: true` in my `~/.stack/config.yaml` anyway.

**Comment by fmthoma** [comment]
See corresponding pull request #2826.

**Comment by fmthoma** [comment]
Closed with 55268fb1bd93e16dabf03807e016785d57169840.
